.github/workflows/test.yaml

```
@@ -0,0 +1,30 @@
+name: test
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  install-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
```

```suggestion
          python-version: "3.13"
```
I'd go up in version number unless there's a specific reason to use 3.11